x-pack/packages/kbn-elastic-assistant-common/impl/data_anonymization/helpers/index.ts

```diff
@@ -6,23 +6,34 @@
  */
 
 import { Replacements } from '../../schemas';
+import { AnonymizationFieldResponse } from '../../schemas/anonymization_fields/bulk_crud_anonymization_fields_route.gen';
 
 export const getIsDataAnonymizable = (rawData: string | Record<string, string[]>): boolean =>
   typeof rawData !== 'string';
 
-export const isAllowed = ({ allowSet, field }: { allowSet: Set<string>; field: string }): boolean =>
-  allowSet.has(field);
+export const isAllowed = ({
+  anonymizationFields,
+  field,
+}: {
+  anonymizationFields: AnonymizationFieldResponse[];
+  field: string;
+}): boolean => anonymizationFields.find((a) => a.field === field)?.allowed ?? false;
 
-export const isDenied = ({ allowSet, field }: { allowSet: Set<string>; field: string }): boolean =>
-  !allowSet.has(field);
+export const isDenied = ({
+  anonymizationFields,
+  field,
+}: {
+  anonymizationFields: AnonymizationFieldResponse[];
+  field: string;
+}): boolean => !(anonymizationFields.find((a) => a.field === field)?.allowed ?? false);
```

Comment:
```suggestion
export const isDenied = (args: {
  anonymizationFields: AnonymizationFieldResponse[];
  field: string;
}): boolean => !isAllowed(args);
```
nit. you could also create a common type to use between these 3 functions like:

```
type FieldAllowedCheck = (args: {
  anonymizationFields: AnonymizationFieldResponse[];
  field: string;
}) => boolean;
```